WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+
+WSL 2 automatically resizes these VHD files to meet storage needs. By default each VHD file used by WSL 2 is initially allocated a 1TB maximum amount of disk space (prior to [WSL release 0.58.0](https://github.com/microsoft/WSL/releases/tag/0.58.0) this default was set to a 256GB maximum).
+If the storage space required by your Linux files exceeds this maximum size, you will see errors stating that you've run out of disk space. To fix this error, follow the guidance below on [How to expand the size of your WSL 2 Virtual Hard Disk](#how-to-expand-the-size-of-your-wsl-2-virtual-hard-disk).
+
+## How to check your available disk space
+
+Check the amount of disk space available in the VHD for a Linux distribution installed with WSL 2 by using the Linux `Df` command.
+
+To check available disk space, open a PowerShell command line and enter this command (replacing `<distribution-name>` with the actual distribution name):
+
+```powershell
+wsl.exe --system -d <distribution-name> df -h /mnt/wslg/distro
+```
+
+The output will include:
+
+- **Filesystem**: Identifier for the VHD file system
+- **Size**: Total size of the disk (the maximum amount of space allocated to the VHD)
+- **Used**: Amount of space currently being used in the VHD
+- **Avail**: Amount of space left in the VHD (Allocated size minus amount used)
+- **Use%**: Percentage of disk space remaining (Used / Allocated size)
+- **Mounted on**: Directory path where the disk is mounted
+
+If you see that you are near to reaching the available amount of disk space allocated to your VHD, or have already received an error due to no disk space remaining, see the next section for steps on how to expand the maximum amount of disk space allocated to the VHD associated with your Linux distribution.
+
+> [!NOTE]
+> The size of the VHD represented by the output of the df command represents the size allocated for the Linux file system and will be different than the total size on your device used by the Linux distribution. To view the **Total usage** size of a Linux distribution installed in your `LocalApps` directory, you can use the Settings app on Windows. Go to **Settings > Apps > Installed apps** and search the list of installed apps by entering the name of the Linux distribution you want to view, for example "Ubuntu" or "Debian". By selecting `...`, you can choose **Advanced options** to view the total disk space used, amount of data stored, and version number of your installed distribution. You can also **Uninstall** the distribution.
```

Comment:
This is necessarily true since the distribution VHDX isn't guaranteed to be stored inside the application's LocalState folder. I'd recommend removing this note

Reply:
Can you clarify what **Settings > Apps > Installed apps > Ubuntu > Advanced options > Total disk space** is defining? Seems worth clarifying in the docs since it is available and discoverable. Does the edit below look correct?

> The size of the VHD represented by the output of the df command represents the size allocated for the Linux file system and will be different than the total size on your device used by the Linux distribution. You can view total disk space used by the distribution by going to **Settings > Apps > Installed apps** and search the list of installed apps by entering the name of the Linux distribution you want to view, for example "Ubuntu" or "Debian". By selecting `...`, you can choose **Advanced options** to view total disk space used, amount of data stored, and version number of your installed distribution. You can also **Uninstall** the distribution.

Reply:
`Settings > Apps > Installed apps > Ubuntu > Advanced options > Total disk space` measure the amount of the storage used by the MSIX application, which can, but isn't guaranteed to contain the distro VHDX

Reply:
You're right -- that gets too confusing. Removing this note. Will reassess if we get any feedback, but otherwise agreeing that including this may just add more confusion.